torch/_functorch/partitioners.py

```diff
@@ -533,12 +533,13 @@ def reordering_to_mimic_autograd_engine(gm: fx.GraphModule) -> fx.GraphModule:
 
     # Populate depth for the nodes. Depth is the distance from the inputs.
     depths = {}
-    output_node = next(iter(gm.graph.find_nodes(op="output")))
     for node in gm.graph.nodes:
         if node.op == "placeholder":
             depths[node] = 0
         else:
-            depths[node] = max([depths[arg] for arg in node.all_input_nodes], default=0)
+            depths[node] = (
+                max((depths[arg] for arg in node.all_input_nodes), default=0) + 1
```

Comment:
This is a bug fix?

Reply:
Yeah 🤔

Reply:
Why it was even working in the first place then? 🤔

Reply:
This is not needed - it's just purely a heuristic to order things a bit nicer.

Reply:
Yeah, +1 was present in the very first PR - https://github.com/pytorch/pytorch/pull/103357/files#diff-6b6c6c8ae7a328efb4e34a103f5886f33b406efab963d6f7330de867638eba40R375-R398

This is very important for manual torch.compile + AC work otherwise the memory footprint because tag-based impl creates duplicated nodes at very different places than what eager would do.

Reply:
@anijain2305 this is actually only a *sub* heuristic within the larger heuristic.

https://github.com/pytorch/pytorch/pull/103357/files#diff-6b6c6c8ae7a328efb4e34a103f5886f33b406efab963d6f7330de867638eba40R456

Like, you're still going to iterate in order in the backwards adding the relevant forward ops first. The depths only affects which arg you add first.